ceph/rados/core_workflows.py

```diff
@@ -725,14 +725,24 @@ def enable_balancer(self, **kwargs) -> bool:
                 f"{mgr_modules['enabled_modules']}"
             )
 
-        # Setting the mode for the balancer. Available modes: none|crush-compat|upmap
-        balancer_mode = kwargs.get("balancer_mode", "upmap")
-        cmd = f"ceph balancer mode {balancer_mode}"
-        self.node.shell([cmd])
         # Turning on the balancer on the system
         cmd = "ceph balancer on"
         self.node.shell([cmd])
 
+        # Setting the mode for the balancer. Available modes: none|crush-compat|upmap|upmap-read|read
+        balancer_mode = kwargs.get("balancer_mode", "upmap")
+        cmd = f"ceph balancer mode {balancer_mode}"
+        self.node.shell([cmd])
+
+        cmd = "ceph balancer status"
```

Comment:
If the mode is being changed, does there need to be a slight delay before validation?

Reply:
Added delay.